src/tests/ftest/security/pool_security_acl.py

```diff
@@ -0,0 +1,400 @@
+            daos_cmd.request.value = "container"
+            daos_cmd.action.value = "create --svc={} --pool={}".format(
+                svc, uuid)
+        elif action.lower() == "read":
+            daos_cmd.request.value = "pool"
+            daos_cmd.action.value = "query --svc={} --pool={}".format(
+                svc, uuid)
+        else:
+            self.fail("##In verify_pool_readwrite, invalid action: "
+                      "%s", action)
+        daos_cmd.exit_status_exception = False
+        result = daos_cmd.run()
+        self.log.info("  In verify_pool_readwrite %s.\n =daos_cmd.run()"
+                      " result:\n%s", action, result)
+        self.verify_daos_pool_result(result, action, expect, deny_access)
+
+    def create_pool_acl(self, num_user, num_group, current_user_acl, acl_file):
+        '''
+        Deascription:
+            Create pool get-acl file with specified number of users and groups
+            with random permission.
+        Args:
+            num_user:  number of user with random permission to be created.
+            num_group: number of group with random permission to be created.
+            current_user_acl: acl entries on current user to be created.
+            acl_file: acl file to be created.
+        Return:
+            permission_list: acl permission list on the acl_file.
+        '''
+        user_prefix = self.params.get("user_prefix", "/run/pool_acl/*")
+        user_list = []
+        group_list = []
+        for uid in range(num_user):
+            username = user_prefix + "_tester_" + str(uid + 1)
+            new_user = "A::" + username + "@:" + PERMISSIONS[uid % 4]
+            add_del_user(self.hostlist_clients, "useradd", username)
+            user_list.append(new_user)
+        for gid in range(num_group):
+            groupname = user_prefix + "_testGrp_" + str(gid + 1)
+            new_group = "A:G:" + groupname + "@:" + PERMISSIONS[(gid + 2) % 4]
+            add_del_user(self.hostlist_clients, "groupadd", groupname)
+            group_list.append(new_group)
+        permission_list = group_list + user_list + current_user_acl
+        random.shuffle(permission_list)
+        with open(acl_file, "w") as test_file:
+            test_file.write("\n".join(permission_list))
+        return permission_list
+
+    def cleanup_user_group(self, num_user, num_group):
+        '''
+        Deascription:
+            Remove the daos acl user and group on host.
+        Args:
+            num_user: number of user to be cleaned.
+            num_group: number of group name to be cleaned.
+        Return:
+            none.
+        '''
+        for user in range(num_user):
+            username = "daos_ci_tester_" + str(user + 1)
+            add_del_user(self.hostlist_clients, "userdel", username)
+        for group in range(num_group):
+            groupname = "daos_ci_testGrp_" + str(group + 1)
+            add_del_user(self.hostlist_clients, "groupdel", groupname)
```

Comment:
Seems like the user and group names are generated using different algorithms here vs. in the create_pool_acl function above. Might make sense to generate the list of names in a helper function that could be used by both.

Reply:
Thanks for the catch, please see the new script with updates.